libraries/classes/Export/TemplateModel.php

```
@@ -0,0 +1,170 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpMyAdmin\Export;
+
+use PhpMyAdmin\DatabaseInterface;
+use PhpMyAdmin\Util;
+use function sprintf;
+
+final class TemplateModel
+{
+    /** @var DatabaseInterface */
+    private $dbi;
+
+    public function __construct(DatabaseInterface $dbi)
+    {
+        $this->dbi = $dbi;
+    }
+
+    /** @return bool|string */
+    public function create(string $db, string $table, Template $template)
+    {
+        $query = sprintf(
+            'INSERT INTO %s.%s (`username`, `export_type`, `template_name`, `template_data`)'
+                . ' VALUES (\'%s\', \'%s\', \'%s\', \'%s\');',
+            Util::backquote($db),
+            Util::backquote($table),
+            $this->dbi->escapeString($template->getUsername()),
+            $this->dbi->escapeString($template->getExportType()),
+            $this->dbi->escapeString($template->getName()),
+            $this->dbi->escapeString($template->getData())
+        );
+        $result = @$this->dbi->tryQuery(
```

why do we need `@` ?

---

It suppress errors. It comes from `Relation::queryAsControlUser`. I will investigate why this operator is there though.